src/java.base/share/classes/jdk/internal/foreign/NativeMemorySegmentImpl.java

```diff
@@ -147,6 +150,14 @@ public void cleanup() {
         return segment;
     }
 
+    private static long allocateMemoryWrapper(long size) {
+        try {
+            return UNSAFE.allocateMemory(size);
+        } catch (IllegalArgumentException ex) {
+            throw new OutOfMemoryError();
```

Comment:
Maybe include the cause?
```suggestion
            throw new OutOfMemoryError(ex);
```

Reply:
OutOfMemoryError doesn't have a constructor that takes an exception. I'll leave it as is for the time being.